### PR TITLE
Remove pickle and replace with json

### PR DESCRIPTION
**Pull request by sanders41**
Closes [#10](https://github.com/ethyca/security-issues/issues/10)

### Code Changes

* Replaces pickle with json encoding for cache.

### Steps to Confirm

* [ ] Create a privacy request
* [ ] Approve and run the privacy request
* [ ] Test any other areas that caches data, the more you can think of the better here.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

When putting this together I ran across a lot of edge cases. This will need a good review to ensure I caught them all.

### CODE REVIEW

**Comment by sanders41** [comment]
Tests are currently failing with `container for service "fides" is unhealthy` before the tests ever start so I get no test results. Containers spin up and tests are running locally so it seems to be something specific to github actions at the moment. Going to try to restart them tomorrow.

**Comment by sanders41** [comment]
@ThomasLaPiana any thoughts here? When I run the tests that are failing here they all pass locally (except mssql, but this is expected). The failures here are because docker is unhealthy, it doesn't seem to even get to the tests.

File: src/fides/api/ops/util/cache.py [hunk 1]
```diff
@@ -20,6 +23,52 @@
...
+            return f"quote_encoded{quote(o)}"
+        if isinstance(o, (datetime, date)):
+            return o.isoformat()
+        if isinstance(o, dict):
+            for _, v in o.items():
+                if not isinstance(v, str) and not isinstance(v, bytes):
+                    self.default(v)
+            return o
+        if isinstance(o, ObjectId):
+            return f"encoded_object_id{str(o)}"
```
Comment (seanpreston) [inline src/fides/api/ops/util/cache.py:42]: This looks like it's to handle the Mongo edge case described below?

File: tests/ops/integration_tests/test_mongo_task.py [hunk 2]
```diff
@@ -53,9 +53,7 @@ async def test_combined_erasure_task(
     """Includes examples of mongo nested and array erasures"""
     policy = erasure_policy("A", "B")
     seed_email = postgres_inserts["customer"][0]["email"]
-    privacy_request = PrivacyRequest(
-        id=f"test_sql_erasure_task_{random.randint(0, 1000)}"
-    )
+    privacy_request = PrivacyRequest(id=f"test_sql_erasure_task_{uuid4()}")
```
Comment (seanpreston) [inline tests/ops/integration_tests/test_mongo_task.py:56]: Thanks, nice change

File: src/fides/api/ops/util/cache.py (hunk 1 above)
Comment (sanders41) [reply src/fides/api/ops/util/cache.py:42]: Correct

**Review by seanpreston — CHANGES_REQUESTED** [review]
Looks mostly good but a couple of minor suggestions, and some Python3.8 tests failing.

**Comment by sanders41** [comment]
@seanpreston thanks for noticing the failures were all 3.8. With that I found the issue.

**Review by seanpreston — APPROVED** [review]
Thanks Paul, good change.
